sys-kernel/scx/scx-9999.ebuild

```diff
@@ -0,0 +1,121 @@
+# Copyright 1999-2024 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LLVM_COMPAT=( 17 18 )
+
+inherit linux-info llvm-r1 cargo meson git-r3
+
+DESCRIPTION="sched_ext schedulers and tools"
+HOMEPAGE="https://github.com/sched-ext/scx"
+EGIT_REPO_URI="https://github.com/sched-ext/scx"
+
+BPFTOOL_REPO_URI="https://github.com/libbpf/bpftool"
+LIBBPF_REPO_URI="https://github.com/libbpf/libbpf"
+
+# we need to bundle bpftool for skeleton support
+# these are just for the build, scx scheds will link
+# to system libbpf
+BPFTOOL_COMMIT="20ce6933869b70bacfdd0dd1a8399199290bf8ff"
+LIBBPF_COMMIT="20ea95b4505c477af3b6ff6ce9d19cee868ddc5d"
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="debug openrc systemd"
+
+BDEPEND="
+	app-misc/jq
+	sys-kernel/linux-headers
+	virtual/pkgconfig
+	$(llvm_gen_dep '
+		sys-devel/clang:${LLVM_SLOT}[llvm_targets_BPF]
+		virtual/rust:0/llvm-${LLVM_SLOT}
+	')
+"
+DEPEND="
+	>=dev-libs/libbpf-1.3.0
+	>=dev-util/bpftool-6.8
+	openrc? ( sys-apps/openrc )
+	systemd? ( sys-apps/systemd )
+"
+RDEPEND="${DEPEND}"
+
+CONFIG_CHECK="
+	~BPF
+	~BPF_EVENTS
+	~BPF_JIT
+	~BPF_SYSCALL
+	~DEBUG_KERNEL
+	~DEBUG_INFO_BTF
+	~FTRACE
+	~SCHED_CLASS_EXT
+"
+COMMON_WARN_BPF="
+The following kernel config settings need to be enabled
+in order to load userspace schedulers:
+	${CONFIG_CHECK//\~/CONFIG_}"
+WARNING_BPF="${COMMON_WARN_BPF}"
+WARNING_BPF_EVENTS="${COMMON_WARN_BPF}"
+WARNING_BPF_JIT="${COMMON_WARN_BPF}"
+WARNING_BPF_SYSCALL="${COMMON_WARN_BPF}"
+WARNING_DEBUG_KERNEL="${COMMON_WARN_BPF}"
+WARNING_DEBUG_INFO_BTF="${COMMON_WARN_BPF}"
+WARNING_FTRACE="${COMMON_WARN_BPF}"
+WARNING_SCHED_CLASS_EXT="
+Make sure your kernel includes the sched-ext
+patchset and enables SCHED_CLASS_EXT!
+
+Kernels including this are:
+	sys-kernel/cachyos-kernel (USE cachyos or sched-ext)
+"
+
+# default but already needed in src_prepare
+BUILD_DIR="${WORKDIR}/${P}-build"
+
+src_unpack() {
+	git-r3_src_unpack
+	shopt -s globstar
+	for manifest in ${WORKDIR}/${P}/**/Cargo.toml; do
+		cargo fetch --manifest-path="$manifest"
```

Comment:
Please use `cargo_live_src_unpack` instead e.g.
```bash
cd "${manifest%/*}" || die
cargo_live_src_unpack
 ```
 
 Might need `cargo_src_configure --frozen` in `src_configure` too according to [cargo.eclass](https://devmanual.gentoo.org/eclass-reference/cargo.eclass/index.html)

Reply:
Oh true, I didn't catch that line. I guess `S="${manifest%/*}" cargo_live_src_unpack` then?
Didn't test it but that should only set S for the single function call while keeping global S intact, no?

Reply:
The reason why I'd prefer cargo_live_src_unpack is that it stores the crates for future use instead of fetching everything at every rebuild (which likely happens a lot with live ebuild)

Reply:
> Oh true, I didn't catch that line. I guess `S="${manifest%/*}" cargo_live_src_unpack` then? Didn't test it but that should only set S for the single function call while keeping global S intact, no?

To keep it intact I used
```bash
(
    S="${manifest%/*}"
    cargo_live_src_unpack
)
``` 
but the issue is the build fails at compile time

Reply:
Can you post what fails? Might be because meson invokes cargo which is different from cargo_src_compile

Reply:
> Can you post what fails? Might be because meson invokes cargo which is different from cargo_src_compile

```
FAILED: rust/scx_utils/Cargo.toml.__PHONY__ 
/usr/lib/python-exec/python3.13/meson --internal exe --unpickle /var/tmp/portage/sys-kernel/scx-9999/work/scx-9999-build/meson-private/meson_exe_cargo_6b8d40ec7587ea41ce9bfdbd34abc79226bfe81f.dat
while executing ['/usr/bin/cargo', 'build', '--manifest-path=../scx-9999/rust/scx_utils/Cargo.toml', '--target-dir=rust/scx_utils', '--quiet', '--release', '--offline']
--- stdout ---

--- stderr ---
error: no matching package named `bindgen` found
location searched: registry `crates-io`
required by package `scx_utils v0.8.1 (/var/tmp/portage/sys-kernel/scx-9999/work/scx-9999/rust/scx_utils)`
As a reminder, you're using offline mode (--offline) which can sometimes cause surprising resolution failures, if this error is too confusing you may wish to retry without the offline flag.

[34/42] /usr/lib/python-exec/python3.13/meson --internal exe --unpickle /var/tmp/portage/sys-kernel/scx-9999/work/scx-9999-build/meson-private/meson_exe_cargo_0f616b195f6790ef6a3b4668051c9abf43728789.dat
FAILED: rust/scx_rustland_core/Cargo.toml.__PHONY__ 
/usr/lib/python-exec/python3.13/meson --internal exe --unpickle /var/tmp/portage/sys-kernel/scx-9999/work/scx-9999-build/meson-private/meson_exe_cargo_0f616b195f6790ef6a3b4668051c9abf43728789.dat
while executing ['/usr/bin/cargo', 'build', '--manifest-path=../scx-9999/rust/scx_rustland_core/Cargo.toml', '--target-dir=rust/scx_rustland_core', '--quiet', '--release', '--offline']
--- stdout ---

--- stderr ---
error: no matching package named `libbpf-rs` found
location searched: registry `crates-io`
required by package `scx_rustland_core v0.5.0 (/var/tmp/portage/sys-kernel/scx-9999/work/scx-9999/rust/scx_rustland_core)`
As a reminder, you're using offline mode (--offline) which can sometimes cause surprising resolution failures, if this error is too confusing you may wish to retry without the offline flag.

[35/42] /usr/lib/python-exec/python3.13/meson --internal exe --unpickle /var/tmp/portage/sys-kernel/scx-9999/work/scx-9999-build/meson-private/meson_exe_cargo_d6add13f5845cd82a0c39ae69e256e2b44bf7634.dat
FAILED: scheds/rust/scx_layered/Cargo.toml.__PHONY__ 
/usr/lib/python-exec/python3.13/meson --internal exe --unpickle /var/tmp/portage/sys-kernel/scx-9999/work/scx-9999-build/meson-private/meson_exe_cargo_d6add13f5845cd82a0c39ae69e256e2b44bf7634.dat
while executing ['/usr/bin/cargo', 'build', '--manifest-path=../scx-9999/scheds/rust/scx_layered/Cargo.toml', '--target-dir=scheds/rust/scx_layered', '--quiet', '--release', '--offline']
--- stdout ---

--- stderr ---
error: no matching package named `bitvec` found
location searched: registry `crates-io`
required by package `scx_layered v0.0.9 (/var/tmp/portage/sys-kernel/scx-9999/work/scx-9999/scheds/rust/scx_layered)`
As a reminder, you're using offline mode (--offline) which can sometimes cause surprising resolution failures, if this error is too confusing you may wish to retry without the offline flag.
```

Reply:
Pushed your workaround, thanks for the help